firefox-ios/Client/Frontend/Autofill/AutofillAccessoryView.swift

```diff
@@ -0,0 +1,112 @@
+/// - `tappedAccessoryButton()`: Handles the tap action on the accessory view.
+class AutofillAccessoryView: UIBarButtonItem {
+    // MARK: - Constants
+    private struct UX {
+        static let accessoryImageViewSize: CGFloat = 24
+        static let accessoryButtonStackViewSpacing: CGFloat = 2
+        static let cornerRadius: CGFloat = 4
+        static let fontSize: CGFloat = 16
+    }
+
+    // MARK: - Properties
+    private let accessoryImageView: UIImageView
+    private let useAccessoryTextLabel: UILabel
+    private let tappedAccessoryButtonAction: (() -> Void)?
+
+    /// Tint color for the accessory image view.
+    var accessoryImageViewTintColor: UIColor? {
+        get {
+            return accessoryImageView.tintColor
+        }
+        set {
+            accessoryImageView.tintColor = newValue
+        }
+    }
+
+    /// Background color for the accessory view.
+    var backgroundColor: UIColor? {
+        didSet {
+            updateBackgroundColor()
+        }
+    }
+
+    // MARK: - Initialization
+    /// Initializes the accessory view with an image, label, and optional tap action.
+    /// - Parameters:
+    ///   - image: The image for the accessory image view.
+    ///   - labelText: The text for the accessory view label.
+    ///   - tappedAction: The closure to be executed when the accessory view is tapped.
+    init(image: UIImage?, labelText: String, tappedAction: (() -> Void)? = nil) {
+        self.accessoryImageView = .build { imageView in
+            imageView.image = image?.withRenderingMode(.alwaysTemplate)
+            imageView.contentMode = .scaleAspectFit
+            imageView.accessibilityElementsHidden = true
+
+            NSLayoutConstraint.activate([
+                imageView.widthAnchor.constraint(equalToConstant: UX.accessoryImageViewSize),
+                imageView.heightAnchor.constraint(equalToConstant: UX.accessoryImageViewSize)
+            ])
+        }
+
+        self.useAccessoryTextLabel = .build { label in
+            label.font = DefaultDynamicFontHelper.preferredFont(withTextStyle: .title3, size: UX.fontSize, weight: .medium)
+            label.text = labelText
+            label.numberOfLines = 0
+            label.accessibilityTraits = .button
+        }
+
+        self.tappedAccessoryButtonAction = tappedAction
+        super.init()
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Setup
+    private func setup() {
+        let stackViewTapped = UITapGestureRecognizer(target: self, action: #selector(tappedAccessoryButton))
+        let accessoryView = UIStackView(arrangedSubviews: [accessoryImageView, useAccessoryTextLabel])
+        accessoryView.spacing = UX.accessoryButtonStackViewSpacing
+        accessoryView.distribution = .equalCentering
+        accessoryView.layer.cornerRadius = UX.cornerRadius
+        accessoryView.addGestureRecognizer(stackViewTapped)
```

Comment:
do we want to add an accessibility identifier and label to this view? cc: @thatswinnie

Reply:
I saw your update on the accessibility label! Either works, it might be more of my preference, since usually I like to add accessibility on the parent view because it encompasses both the image + label since they aren't distinctive in this case. However, @thatswinnie might have better input on accessibility.